src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -210,17 +216,47 @@ internal HttpConnection(HttpConnectionOptions httpConnectionOptions, ILoggerFact
 
         private async Task SelectAndStartTransport(TransferFormat transferFormat)
         {
-            if (_httpConnectionOptions.Transports == HttpTransportType.WebSockets)
+            var uri = _httpConnectionOptions.Url;
+            // Set the initial access token provider back to the original one from options
+            _accessTokenProvider = _httpConnectionOptions.AccessTokenProvider;
+
+            if (_httpConnectionOptions.SkipNegotiation && _httpConnectionOptions.Transports == HttpTransportType.WebSockets)
```

Comment:
We should either throw if `SkipNegotiation` and `Transports != HttpTransportType.WebSockets` or just be prepared for that bug report to come in: "I set SkipNegotiation and it still sends a negotiate request" ;). I really don't mind which, just saying.